src/app/pcasts/dao/feed_dao.py

```diff
@@ -0,0 +1,170 @@
+  result, heap, count = [], [], 0
+  for source_idx, s in enumerate(sources):
+    if s:
+      item = s.pop()
+      heapq.heappush(heap, (-time.mktime(item.created_at.timetuple()),
+                            item, source_idx))
+  while len(heap) > 0 and count < page_size:  #pylint: disable=C1801
+    _, item, source_idx = heapq.heappop(heap)
+    result.append(FeedElement(item, contexts[source_idx]))
+    if sources[source_idx]:
+      raw_content = sources[source_idx].pop()
+      heapq.heappush(heap, (-time.mktime(raw_content.created_at.timetuple()),
+                            raw_content, source_idx))
+    count += 1
+  return result
+
+def attach_fields_to_json(feed_element, feed_element_json, user):
+  if feed_element.context == FeedContexts.FOLLOWING_RECOMMENDATION:
+    feed_element_json['context_supplier']['is_following'] = users_dao.\
+        is_following_user(
+            user.id,
+            feed_element_json['context_supplier']['id']
+        )
+    feed_element_json['content']['series']['is_subscribed'] = series_dao.\
+        is_subscribed_by_user(
+            feed_element_json['content']['series']['id'],
+            user.id
+        )
+  elif feed_element.context == FeedContexts.FOLLOWING_SUBSCRIPTION:
+    feed_element_json['context_supplier']['is_following'] = users_dao.\
+        is_following_user(
+            user.id,
+            feed_element_json['context_supplier']['id']
+        )
+  elif feed_element.context == FeedContexts.NEW_SUBSCRIBED_EPISODE:
+    feed_element_json['content']['series']['is_subscribed'] = series_dao.\
+        is_subscribed_by_user(
+            feed_element_json['content']['series']['id'],
+            user.id
+        )
+
+  return feed_element_json
+
+def get_following_subscriptions(user_id, maxtime, page_size):
+  followings = followings_dao.get_followings(user_id)
+  following_ids = [f.followed_id for f in followings]
+  maxdatetime = datetime.datetime.fromtimestamp(int(maxtime))
+  subscriptions = Subscription.query \
+    .filter(Subscription.user_id.in_(following_ids),
+            Subscription.created_at <= maxdatetime) \
+    .order_by(Subscription.created_at.desc()) \
+    .limit(page_size) \
+    .all()
+  series = series_dao.\
+    get_multiple_series([s.series_id for s in subscriptions], user_id)
+
+  series_id_to_series = {s.id:s for s in series}
+  for sub in subscriptions:
+    sub.series = series_id_to_series[sub.series_id]
+
+  return subscriptions
+
+def get_following_recommendations(user_id, maxtime, page_size):
+  followings = followings_dao.get_followings(user_id)
+  following_ids = [f.followed_id for f in followings]
+  maxdatetime = datetime.datetime.fromtimestamp(int(maxtime))
+  recommendations = Recommendation.query \
+    .filter(Recommendation.user_id.in_(following_ids),
+            Recommendation.created_at <= maxdatetime) \
+    .order_by(Recommendation.created_at.desc()) \
+    .limit(page_size) \
+    .all()
+  episodes = episodes_dao.\
```

Comment:
Do these indents match out general style (4 spaces after 80+char line vs 2 spaces). This uses both